lib/Controller.js

```diff
@@ -226,16 +235,33 @@ module.exports = (function(){
                     processNoLongerLivePhantomEntries.call(that, liveEntries);
                     processNoLongerLiveEntries.call(that, liveEntries);
                 }
-            }).catch(function(err) {
+            })
+            .catch(function(err) {
                 logger.warn("Failed to execute handleEntries with error [%s]", ErrorUtils.error2string(err));
             });
     }
 
     Controller.prototype.start = function() {
         var that = this;
+
+        this.once('exit', function(code) {
+            if (that.unitTestEventsHandler) {
+                that.unitTestEventsHandler.emit('exit', code);
+            } else {
+                process.exit(code);
+                // Todo: check if a bug --> stop() doesn't work as expected
+                // that.stop();
+            }
+        });
+        adapter = adapter.getAdapter(that);
+
         logger.warn("Starting liveProxy pid: %s, wowza: '%s', API server: '%s', prefix='%s'",process.pid,config.get("mediaServer").hostname,config.get('backendClient').serviceUrl,that.prefix);
         logger.warn("Log files %s",config.get('logFileName'));
         return Q.fcall(function() {
+            if (regressionConfig.command && regressionConfig.command.record ||
```

Comment:
please move it from here... (same comment from above) you already have Controller wrapper